src/compiler/compile/render_dom/wrappers/Element/index.ts

```diff
@@ -1015,6 +1021,41 @@ function to_html(wrappers: Array<ElementWrapper | TextWrapper | MustacheTagWrapp
 			if (!wrapper.void) {
 				state.quasi.value.raw += '>';
 
+				if (wrapper.node.name === 'pre') {
+					// Two or more leading newlines are required to restore the leading newline immediately after `<pre>`.
+					// see https://html.spec.whatwg.org/multipage/grouping-content.html#the-pre-element
+					const first = wrapper.fragment.nodes[0];
+					if (first && first.node.type === 'Text' && start_newline.test(first.node.data)) {
+						state.quasi.value.raw += '\n';
+					}
+				}
+
+				if (is_empty_textarea) {
+					// The <textarea> renders the value attribute as content because the content is stored in the value attribute.
+					const value_attribute = wrapper.attributes.find(attr => attr.node.name === 'value');
+					if (value_attribute) {
+						// Two or more leading newlines are required to restore the leading newline immediately after `<textarea>`.
+						// see https://html.spec.whatwg.org/multipage/syntax.html#element-restrictions
+						const first = value_attribute.node.chunks[0];
+						if (first && first.type === 'Text' && start_newline.test(first.data)) {
+							state.quasi.value.raw += '\n';
+						}
+						value_attribute.node.chunks.forEach(chunk => {
+							if (chunk.type === 'Text') {
+								state.quasi.value.raw += escape_html(chunk.data);
+							} else {
+								literal.quasis.push(state.quasi);
+								literal.expressions.push(chunk.manipulate(block));
+
+								state.quasi = {
+									type: 'TemplateElement',
+									value: { raw: '' }
+								};
+							}
+						});
+					}
```

Comment:
should be able to reuse `to_html` ?

something like `to_html(value_attribute.node.chunks, block, literal, state)` ?

Reply:
I don't think we can use `to_html`, because `value_attribute.node.chunks` is an array of not wrapper nodes.

But since it's the same as L1004, do you think I should add a function and use it?
https://github.com/sveltejs/svelte/pull/7280/files#diff-d6a6e8b36b1b88c9bd08d5bee1887bfffc7cebb2211ce45591ef59f8d2fc1111R1004

Reply:
oh yes 🙂